util/include/cppmicroservices/util/BundleMachOFile.h

```diff
@@ -71,6 +72,35 @@ I readBE(I i)
 #endif
 }
 
+template<typename SegmentCommand, typename Section>
+std::shared_ptr<RawBundleResources> GetBundleContainer(std::ifstream& fs, std::size_t fileOffset, uint32_t lcmd_offset)
+{
+  fs.seekg(fileOffset + lcmd_offset);
+  SegmentCommand segment;
+  fs.read(reinterpret_cast<char*>(&segment), sizeof(SegmentCommand));
+  if(0 == strcmp("__TEXT", segment.segname)) {
+    // find "us_resources" section
+    for (uint32_t i = 0; i < segment.nsects; ++i) {
+      Section section;
+      fs.read(reinterpret_cast<char*>(&section), sizeof(Section));
+      if (0 == strcmp("us_resources", section.sectname)) {
+        uint64_t zipSize = section.size;
+        if (0 < zipSize) {
+          void* zipData = malloc(zipSize * sizeof(char));
```

Comment:
```suggestion
          std::unique_ptr<char[]> zipData = std::unique_ptr(new char[zipSize]);
          if(zipData) {
            fs.seekg(fileOffset + section.offset);
            fs.read(zipData.get(), zipSize);
            return std::make_shared<RawBundleResources>( std::move(zipData), zipSize );
          }
```

Reply:
The miniz api, ```mz_zip_reader_init_mem``` takes a ```void*``` for the data. I chose to use ```malloc```/```free``` so I wouldn't have to to use casts.